src/components/SSOButton/SSOButton.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * This component is used for SSO solution.
+ * Inherit the Button component and only customise the openOauth function, in order to support QG SSO solution.
+ * This component is extending from:
+ * https://github.com/formio/formio.js/blob/master/src/components/button/Button.js
+ *
+ */
+
+import _ from "lodash";
+import NativePromise from "native-promise-only";
+
+const Button = Formio.Components.components.button;
+
+export class SSOButton extends Button {
+  static schema(...extend) {
+    return Button.schema({
+      type: "ssobutton",
+      label: "SSO button",
+      key: "ssobutton",
+      ...extend,
+    });
+  }
+
+  openOauth(settings) {
+    if (!this.root.formio) {
+      console.warn(
+        "You must attach a Form API url to your form in order to use OAuth buttons."
+      );
+      return;
+    }
+
+    /* eslint-disable camelcase */
+    let params = {
+      response_type: "code",
+      client_id: settings.clientId,
+      redirect_uri:
+        (settings.redirectURI && this.interpolate(settings.redirectURI)) ||
+        window.location.origin ||
+        `${window.location.protocol}//${window.location.host}`,
+      state: settings.state,
+      scope: settings.scope,
+    };
+    /* eslint-enable camelcase */
+
+    // Needs for the correct redirection URI for the OpenID
+    const originalRedirectUri = params.redirect_uri;
+
+    // Make display optional.
+    if (settings.display) {
+      params.display = settings.display;
+    }
+
+    params = Object.keys(params)
+      .map((key) => {
+        return `${key}=${encodeURIComponent(params[key])}`;
+      })
+      .join("&");
+
+    const separator = settings.authURI.indexOf("?") !== -1 ? "&" : "?";
+    const url = `${settings.authURI}${separator}${params}`;
+    const popup = window.open(url, settings.provider, "width=1020,height=618");
```

Comment:
is the window width/height ok? could we use the main window for window size?

Reply:
the size seems work well with sso login page so far.
better to leave it for now as the code is inherited from formio's Button, see if they take our upstream code first.